src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -72,8 +72,7 @@ public Task<Socket> AcceptAsync(Socket? acceptSocket)
             {
                 // The operation completed synchronously.  Get a task for it.
                 t = saea.SocketError == SocketError.Success ?
-                    Task.FromResult(saea.AcceptSocket!) :
-                    Task.FromException<Socket>(GetException(saea.SocketError));
+                    Task.FromResult(saea.AcceptSocket!) : TaskToApm.GetSynchronousExceptionTask<Socket>(GetException(saea.SocketError));
```

Comment:
Note this is making the failure path more expensive for everyone, not just code using the APM methods.